### PR TITLE
Feat/allow multiple secrets

### PR DESCRIPTION
**Pull request by psibean**
To address #34

### CODE REVIEW

File: src/tests/testsuite.ts [hunk 1]
```diff
@@ -124,9 +125,56 @@ export const createTestSuite: CreateTestsuite = (name, doubleCsrfOptions) => {
...
+              mockRequest.secret as string
+            )}`)
+          : (mockRequest.cookies[cookieName] = "invalid-value");
+
+        expect(() =>
+          generateToken(mockRequest, mockResponse, false, true)
+        ).to.throw(invalidCsrfTokenError.message);
+      });
+
+      it("should not throw (instead, overwrites wrong token) if csrf cookie is present, it is invalid (wrong token + hash pair, or not a correct value), overwrite is false, and validateOnGeneration is false", () => {
```
Comment (psibean) [inline src/tests/testsuite.ts:145]: should not throw if csrf cookie is present and invalid when overwrite is false, and validateOnGeneration is disabled

File: src/tests/testsuite.ts [hunk 2]
```diff
@@ -112,7 +113,7 @@ export const createTestSuite: CreateTestsuite = (name, doubleCsrfOptions) => {
         assert.notEqual(generatedToken, csrfToken);
       });
 
-      it("should throw if csrf cookie is present, it is invalid (wrong token + hash pair, or not a correct value) and overwrite is false", () => {
+      it("should throw if csrf cookie is present, it is invalid (wrong token + hash pair, or not a correct value), overwrite is false, and validateOnGeneration is true", () => {
```
Comment (psibean) [inline src/tests/testsuite.ts:116]: should throw if csrf cookie is present and invalid, overwrite is false, and validateOnGeneration is enabled

File: src/tests/doublecsrf.test.ts [hunk 3]
```diff
@@ -1,14 +1,239 @@
...
+        signed,
+        generateToken,
+        validateRequest,
+      }),
+      validateRequest,
+      generateToken,
+    };
+  };
+
+  context("validating requests with combination of different secret/s", () => {
```
Comment (psibean) [inline src/tests/doublecsrf.test.ts:71]: TIL about context blocks.!

**Comment by psibean** [comment]
Logic is pretty sound, README update is definitely needed

**Review by davidgonmar — DISMISSED** [review]

File: CHANGELOG.md
Comment (psibean) [inline CHANGELOG.md:1]: Ah we probably don't need this file manually changed.

This is automatically updated / generated as part of the release using the conventional commits changelog generator.

**Pull request merged**